v0/destinations/profitwell/transform.js

```diff
@@ -0,0 +1,294 @@
+      return false;
+    });
+
+    if (!subscriptionFound) {
+      // for a given userId, subscriptionId not found
+      // dropping event if profitwellSubscriptionId (externalId) did not
+      // match with any subscription_id at destination
+      if (subscriptionId) {
+        throw new CustomError("profitwell subscription_id not found", 400);
+      }
+      payload = constructPayload(message, createPayloadMapping);
+      payload = {
+        ...payload,
+        user_id: userId,
+        user_alias: userAlias
+      };
+      if (
+        payload.plan_interval &&
+        !(
+          payload.plan_interval.toLowerCase() === "month" ||
+          payload.plan_interval.toLowerCase() === "year"
+        )
+      ) {
+        throw new CustomError("invalid format for planInterval. Aborting", 400);
+      }
+      if (
+        payload.plan_currency &&
+        !isValidPlanCurrency(payload.plan_currency)
+      ) {
+        payload.plan_currency = null;
+      }
+      if (
+        payload.status &&
+        !(
+          payload.status.toLowerCase() === "active" ||
+          payload.status.toLowerCase() === "trialing"
+        )
+      ) {
+        payload.status = null;
+      }
+      payload.effective_date = unixTimestampOrError(
+        payload.effective_date,
+        message.originalTimestamp
+      );
+      response.method = defaultPostRequestConfig.requestMethod;
+      response.endpoint = `${BASE_ENDPOINT}/v2/subscriptions/`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+
+    // for a given userId, subscription is found at dest.
+    if (valFound) {
+      payload = constructPayload(message, updatePayloadMapping);
+      if (
+        payload.plan_interval &&
+        !(
+          payload.plan_interval.toLowerCase() === "month" ||
+          payload.plan_interval.toLowerCase() === "year"
+        )
+      ) {
+        throw new CustomError("invalid format for planInterval. Aborting", 400);
+      }
+      if (
+        payload.status &&
+        !(
+          payload.status.toLowerCase() === "active" ||
+          payload.status.toLowerCase() === "trialing"
+        )
+      ) {
+        payload.status = null;
+      }
+      payload.effective_date = unixTimestampOrError(
+        payload.effective_date,
+        message.originalTimestamp
+      );
+      response.method = defaultPutRequestConfig.requestMethod;
+      response.endpoint = `${BASE_ENDPOINT}/v2/subscriptions/${subscriptionId ||
+        subscriptionAlias}/`;
+      response.headers = {
+        "Content-Type": "application/json",
+        Authorization: Config.privateApiKey
+      };
+      response.body.JSON = removeUndefinedAndNullValues(payload);
+      return response;
+    }
+  }
+
+  // handler for other destination side errors
+  if (res.response.response.status !== 404) {
```

Comment:
`res.response.response.status` Here res.response is error and hence we are doing `error.response.status`. Can you confirm this?

Reply:
how about just introducing a temp variable `const error = resp.response` ? makes it a bit readable and easier to understand.

Reply:
Refactored